### ISSUE TITLE
Create "All Metrics" guide

### ISSUE THREAD

**Issue by jclingan** [quarkusio/quarkus#10076]
**Description**
The "[all configuration](https://quarkus.io/guides/all-config)" guide is very useful in understanding how Quarkus can be configured. This concept should be extended to metrics, with the creation of an "all metrics" guide that is an aggregation of core quarkus metrics plus the metrics available in all extensions.  This also builds confidence in Quarkus being an observable platform.

The guide can include metric name, description, extension which defines it, metric type (gauge, counter, etc).

**Comment by kenfinnigan** [comment]
Is it purely a reference thing?

Would it be better to have an auto generated dashboard containing the metrics from all extensions within an application?

**Comment by jmartisk** [comment]
There exists a Grafana dashboard generator usable with Quarkus (https://github.com/jamesfalkner/microprofile-grafana) but I'm not sure if all users are going to use Grafana only
It also has the problem that some metrics are registered lazily (and the full list is not known in advance), so running the generator against an application which has just booted might not include all metrics.

**Comment by ebullient** [comment]
This is going to get more complicated as micrometer comes in, as the two metrics systems use different naming conventions. Still possible, just different.

**Comment by jclingan** [comment]
@kenfinnigan I was seeing this as a reference doc, just like the all-config guide. DevOps/SRE teams want to know what metrics are available to them when evaluating Quarkus or even after selecting Quarkus. 

@jmartisk Regarding lazy generation, that could be noted in the reference guide.

@ebullient Yeah, understood.

This is not a high-priority issue but is a reference guide is definitely a nice-to-have guide. I figured at the time I wrote this that it shouldn't be too difficult since we have the metadata we need, but I can see the complications per above.

**Comment by jmartisk** [comment]
@rquinio, are you envisioning any other benefit from this apart from being able to generate documentation automatically?

I'm not sure I understand how the metrics from these annotations would be matched to their underlying implementation/function, for example `Statistics::getSessionOpenCount` for the `hibernate-orm.sessions.open` metric? Reference to the `Statistics` object is only obtained at runtime, so I guess you wouldn't be able to declare that from the annotation directly.

Per your code snippet, we would also have to create new abstraction classes for `Counter` and other metric types that are implementation-neutral (to be able to make this work with any Metrics implementation (Micrometer, for example)), that makes it somewhat complicated

**Comment by rquinio** [comment]
@jmartisk No, I don't actually see other benefits, so it may not be worth it.

Sorry I should have put the corresponding runtime init snippet:

```java
void registerMetrics(MetricsFactory metricsFactory, HibernateOrmMetrics ormsMetrics, String puName, Statistics statistics) {
   metricsFactory.builder(ormsMetrics.sessionMetrics.open) //Initialize from metric metadata (name, type, description, tags)
                .tag("entityManagerFactory", puName)
                .buildCounter(statistics, Statistics::getSessionOpenCount); 
}
```
I'm not exactly sure how the metric types should be handled (an attribute of @MetricItem might be enough, if not leveraged by MetricsFactory builders)
